recipes/ginkgo/all/conanfile.py

```diff
@@ -100,29 +109,57 @@ def package_info(self):
         self.cpp_info.names["pkg_config"] = "ginkgo"
 
         debug_suffix = "d" if self.settings.build_type == "Debug" else ""
+        has_dpcpp_device = Version(self.version) >= Version("1.4.0")
 
         self.cpp_info.components["ginkgo_core"].names["cmake_find_package"] = "ginkgo"
         self.cpp_info.components["ginkgo_core"].names["cmake_find_package_multi"] = "ginkgo"
         self.cpp_info.components["ginkgo_core"].names["pkg_config"] = "ginkgo"
-        self.cpp_info.components["ginkgo_core"].libs = ["ginkgo" + debug_suffix]
+        self.cpp_info.components["ginkgo_core"].libs = [
+            "ginkgo" + debug_suffix]
         self.cpp_info.components["ginkgo_core"].requires = [
             "ginkgo_omp", "ginkgo_cuda", "ginkgo_reference", "ginkgo_hip"
         ]
 
         self.cpp_info.components["ginkgo_cuda"].names["cmake_find_package"] = "ginkgo_cuda"
         self.cpp_info.components["ginkgo_cuda"].names["cmake_find_package_multi"] = "ginkgo_cuda"
-        self.cpp_info.components["ginkgo_cuda"].libs = ["ginkgo_cuda" + debug_suffix]
+        self.cpp_info.components["ginkgo_cuda"].libs = [
+            "ginkgo_cuda" + debug_suffix]
         self.cpp_info.components["ginkgo_cuda"].requires = ["ginkgo_hip"]
 
         self.cpp_info.components["ginkgo_omp"].names["cmake_find_package"] = "ginkgo_omp"
         self.cpp_info.components["ginkgo_omp"].names["cmake_find_package_multi"] = "ginkgo_omp"
-        self.cpp_info.components["ginkgo_omp"].libs = ["ginkgo_omp" + debug_suffix]
-        self.cpp_info.components["ginkgo_omp"].requires = ["ginkgo_cuda", "ginkgo_hip"]
+        self.cpp_info.components["ginkgo_omp"].libs = [
+            "ginkgo_omp" + debug_suffix]
+        self.cpp_info.components["ginkgo_omp"].requires = [
+            "ginkgo_cuda", "ginkgo_hip"]
 
         self.cpp_info.components["ginkgo_hip"].names["cmake_find_package"] = "ginkgo_hip"
         self.cpp_info.components["ginkgo_hip"].names["cmake_find_package_multi"] = "ginkgo_hip"
-        self.cpp_info.components["ginkgo_hip"].libs = ["ginkgo_hip" + debug_suffix]
+        self.cpp_info.components["ginkgo_hip"].libs = [
+            "ginkgo_hip" + debug_suffix]
 
         self.cpp_info.components["ginkgo_reference"].names["cmake_find_package"] = "ginkgo_reference"
         self.cpp_info.components["ginkgo_reference"].names["cmake_find_package_multi"] = "ginkgo_reference"
-        self.cpp_info.components["ginkgo_reference"].libs = ["ginkgo_reference" + debug_suffix]
+        self.cpp_info.components["ginkgo_reference"].libs = [
+            "ginkgo_reference" + debug_suffix]
+
+        if has_dpcpp_device:
```

Comment:
If the cmake option is false why are the components being added?

Reply:
Because the libraries are still added and linked as stubs, to be replaced by the real implementation if necessary.

Reply:
Let's leave a comment for the next guy =)

```suggestion
        if has_dpcpp_device: # Always add these components
            # See https://github.com/conan-io/conan-center-index/pull/7044#discussion_r698181588
```